### PR TITLE
Refactor sex related places

### PR DESCRIPTION
**Pull request by cuducos**
Thanks to @guilhermeprokisch we've got a great start in this topic.

This pulls request merges his contributions with some refactor @anaschwendler and I did the other day.

Still a WIP as we have to test it with the proper API key and as we have a new code review — thanks @irio — to take into account.

### CODE REVIEW

**Comment by Irio** [comment]
What's missing here, @cuducos?

**Comment by cuducos** [comment]
Some refactors/improvememts:

- [ ] Avoid loosing data when a request fail (concurrency issue on file writing, partially solved with `Queue` but I need to test it further)
- [ ] Enhance performance with more parallel requests (revert to `asyncio` commit but keep the queue for file writing)
- [ ] :nail_care: Standardize coroutines (discard `grequests` in favor of `asyncio`)

I feel like I still need to work on that since we haven't been able to download ghe full dataset yet. Unfortunately I'm more focused on Jarbas and on documenting Rosie and the toolbox right now.

**Comment by cuducos** [comment]
Finally! This script is ready for code review 🎊 🍾 ✨ 🎉 

Thanks for the great work from @guilhermeprokisch that gave us a base for this data collection, thanks for numerous pair prigramming sessions with @jtemporal and @anaschwendler, we have a neat, fast and modern script collecting data.

I'm submitting it for code review and starting to collect data. As soon as I have the database (probably in six days due to API cap around 150k requests/day) I will:

* Create a sample notebook in this repo
* Upload the dataset do S3 and add it to the tool box

File: src/fetch_sex_places.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+        Use aiohttp to get the closest place to the company according to each
+        keyworn. Returns `None`, the results are saved on self.places.
+        """
+        tasks = [self.load_place(k) for k in self.KEYWORDS]
+        await asyncio.gather(*tasks)
+
+    async def get_closest(self):
+        """
+        Gets the closest place find in reads self.places, queries for its
+        details and returns a dictoinary with all the info on that place.
```
Comment (daneoshiga) [inline src/fetch_sex_places.py]: typo `dictoinary` and i would suggest using just `dict` when referencing the type.

**Comment by cuducos** [comment]
Good news: script is ready. Bad news: dataset is basically worthless.

---

We don't have funds to gather **all** data: due to Google Places API pricing policy wither it would take ages to collect data using the free tier, or it would cost more than 300k USD. Thus the script now takes three filters: `term`, `value` and `city` (check `python src/fetch_sex_places.py --help` for details).

> Don't forget to `pip install -r requirements.txt` as we added `aiohttp` and `aiofiles` to the stack.

According to [a random sample that looked promising](https://twitter.com/cuducos/status/840882495868530688) the aforementioned dataset has data for a couple of cities — check 834ab3c for details.

The important thins however is that we have tons of _false positives_, so I added a notebook commenting on that — also check 834ab3c for details. 

---

Either way I believe these results — the data collection, the brief analysis and the notebook discussing the quality of data — can be merged into `master` as a documentation of the current status of the hypothesis raised in #39.

**Comment by anaschwendler** [comment]
Took some time to look for this PR.
I already knew the algorithm for the solution, and for me it looks great.
For me it look real good to be merged, so if @jtemporal thinks the same, we can merge it.

**Comment by jtemporal** [comment]
I agree with @anaschwendler I'm merging this one ;) great job everyone!

**Comment by guilhermeprokisch** [comment]
To try to find more reliable results. I suggest filtering the companies name in the bills for some typical stooge names, like bars and hotels.

**Comment by guilhermeprokisch** [comment]
But in the end the need a human confirmation that such place is stooge place. For this, we can call them or go there.

**Pull request merged**